.github/copilot-instructions.md

```diff
@@ -0,0 +1,150 @@
+# Copilot Instructions for kotlin-common
+
+## Repository Overview
+
+### Summary
+Huanshankeji Kotlin Common is a collection of common code libraries in Kotlin that extend the Kotlin language and its standard library. It provides extension libraries for various Kotlin and Java libraries including Arrow, Coroutines, Exposed, Ktor, reflection, Serialization, Vert.x, and more.
+
+### High Level Repository Information
+- **Type**: Kotlin Multiplatform library project
+- **Size**: Medium-sized (23 modules across core and extension libraries)
+- **Languages**: Kotlin (primary), Gradle Kotlin DSL for build scripts
+- **Target Platforms**: JVM, JS (browser), iOS (`iosX64`, `iosArm64`, `iosSimulatorArm64`), WebAssembly JS
+- **Build System**: Gradle with custom build plugins
+- **Documentation**: Generated with Dokka
+- **Testing**: Kotlin Test with Kotest property testing
+
+## Build Instructions
+
+### Prerequisites
+- **JDK 8 (Zulu distribution)** - Required for building and testing (as specified in CI workflows)
+- **macOS environment** recommended for full multiplatform support (Linux has limited iOS target support)
+- Initial setup time: ~2-3 minutes for Gradle daemon initialization and dependency download
+
+### Environment Setup
+Always ensure JDK 8 is properly configured before building. The project uses Gradle wrapper, so no manual Gradle installation is needed.
+
+**IMPORTANT**: On non-macOS systems, iOS targets (`iosArm64`, `iosSimulatorArm64`, `iosX64`) will be disabled. Add `kotlin.native.ignoreDisabledTargets=true` to suppress warnings.
+
+### Bootstrap and Build Commands
+
+#### Essential Commands (validated and working):
+1. **Clean the project**: `./gradlew clean` (takes ~2-3 seconds after initial setup)
+2. **Build all modules**: `./gradlew build` (takes 5-10 minutes initially)
+3. **Run verification checks**: `./gradlew check` (comprehensive testing, takes 5-15 minutes)
+4. **Publish to local Maven**: `./gradlew publishToMavenLocal` (recommended for testing changes)
+
+#### Testing Commands:
+- **Run JVM tests for a module**: `./gradlew :kotlin-common-[module]:jvmTest`
+- **Run JS tests for a module**: `./gradlew :kotlin-common-[module]:jsTest` 
+- **Run all tests for a module**: `./gradlew :kotlin-common-[module]:allTests`
+- **Global test command**: There is no single `test` task - use `check` for comprehensive verification
```

Comment:
For a JVM only module (configured with `kotlin("jvm")`, possibly indirectly with the `buildSrc/src/main/kotlin/jvm-conventions.gradle.kts` plugin), the test task is just `test`; for a multiplatform module, the test task to run only the JVM tests is `jvmTest` and the one to run all tests is `allTests`. `check` does cover all tests.